cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CreateStackAction.java

```diff
@@ -1,115 +1,131 @@
+                    @Output(Outputs.RETURN_RESULT),
+                    @Output(Outputs.EXCEPTION)
+            },
+            responses = {
+                    @Response(text = Outputs.SUCCESS, field = Outputs.RETURN_CODE, value = Outputs.SUCCESS_RETURN_CODE,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.RESOLVED),
+                    @Response(text = Outputs.FAILURE, field = Outputs.RETURN_CODE, value = Outputs.FAILURE_RETURN_CODE,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.ERROR)
+            }
+    )
+    public Map<String, String> execute(
+            @Param(value = IDENTITY,   required = true)  String identity,
+            @Param(value = CREDENTIAL, required = true, encrypted = true)  String credential,
+            @Param(value = REGION,     required = true)  String region,
+            @Param(value = PROXY_HOST)                   String proxyHost,
+            @Param(value = PROXY_PORT)                   String proxyPort,
+            @Param(value = PROXY_USERNAME)               String proxyUsername,
+            @Param(value = PROXY_PASSWORD, encrypted = true) String proxyPassword,
+            @Param(value = CONNECT_TIMEOUT)              String connectTimeoutMs,
+            @Param(value = EXECUTION_TIMEOUT)            String execTimeoutMs,
+            @Param(value = STACK_NAME, required = true)  String stackName,
+            @Param(value = TEMPLATE_BODY, required = true) String templateBody,
+            @Param(value = PARAMETERS)                   String parameters) {
+
+        proxyPort = defaultIfEmpty(proxyPort, DefaultValues.PROXY_PORT);
+        connectTimeoutMs = defaultIfEmpty(connectTimeoutMs, DefaultValues.CONNECT_TIMEOUT);
+        execTimeoutMs = defaultIfEmpty(execTimeoutMs, DefaultValues.EXEC_TIMEOUT);
+
+        try {
+            AmazonCloudFormation stackBuilder = CloudFormationClientBuilder.getCloudFormationClient(identity, credential, proxyHost, proxyPort, proxyUsername, proxyPassword, connectTimeoutMs, execTimeoutMs, region);
+
+            CreateStackRequest createRequest = new CreateStackRequest()
+                    .withStackName(stackName)
+                    .withTemplateBody(templateBody)
+                    .withParameters(toArrayOfParameters(parameters));
+
+            CreateStackResult result = stackBuilder.createStack(createRequest);
+
+            return OutputUtilities.getSuccessResultsMap(result.toString());
+        } catch (Exception e) {
+            return OutputUtilities.getFailureResultsMap(e);
+        }
+    }
+
+    private ArrayList<Parameter> toArrayOfParameters(String parameters) {
+        ArrayList<Parameter> parametersList = new ArrayList<Parameter>();
+
+        for (String line : parameters.split("\n")) {
+
+            ParametersLine paramLine = new ParametersLine(line);
+
+            if (!paramLine.isValid()) {
+                continue;

```

Comment:
I'm personally against using **continue** and **break** in a for statement just because I consider it's harder to understand this way. Consider creating and adding a new **Parameter** object in **parametersList** if **paramLine** is valid. This way the **continue** is no longer needed. If the `paramLine.isValid()` method will return false, the for statement will just move to the next element.